lib/IDE/Refactoring.cpp

```diff
@@ -1463,6 +1463,64 @@ bool RefactoringActionExtractRepeatedExpr::performChange() {
                                           EditConsumer).performChange();
 }
 
+struct CollapsibleNestedIfInfo {
+    IfStmt *OuterIf;
+    IfStmt *InnerIf;
+    CollapsibleNestedIfInfo(): OuterIf(nullptr), InnerIf(nullptr) {}
+    bool isValid() { return OuterIf && InnerIf; }
+};
+
+static CollapsibleNestedIfInfo findCollapseNestedIfTarget(ResolvedCursorInfo CursorInfo) {
+    if (CursorInfo.Kind != CursorInfoKind::StmtStart)
+        return CollapsibleNestedIfInfo();
+    struct IfStmtFinder: public SourceEntityWalker {
+        SourceLoc StartLoc;
+        CollapsibleNestedIfInfo IfInfo;
+        IfStmtFinder(SourceLoc StartLoc): StartLoc(StartLoc), IfInfo() {}
+        bool walkToStmtPre(Stmt *S) {
```

Comment:
Nitpick: LLVM style prefers early returns, rather than nesting the body of the function. This would be better expressed by:

```
if (S->getKind() != StmtKind::If) {
  return false;
}
```

Reply:
Nitpick: LLVM style uses 2 spaces for indentation.

Reply:
Could also dyn_cast first

```c++
auto *IFS = dyn_cast<IfStmt>(S);
if (!IFS) return false;
/**/ 
```